File: python/triangle/triangle.py
def validator(func):
  return lambda x: 0 not in x and sum(x) - max(x) > max(x) and func(x)

@validator
def is_equilateral(sides):
  return len(set(sides)) == 1

@validator
def is_isosceles(sides):
  return len(set(sides)) <= 2

@validator
def is_scalene(sides):
  return len(set(sides)) == 3

File: kotlin/triangle/src/main/kotlin/Triangle.kt
class Triangle<T>(val a: T, val b: T, val c: T) 
    where T : Number, T : Comparable<T> {
    
    private val sides: Set<T>

    init {
        val sortedSides = sequenceOf(a, b, c).sorted().toList()
        if (sortedSides[0].toDouble() + sortedSides[1].toDouble() <= sortedSides[2].toDouble()) {
            throw IllegalArgumentException("Not a triangle")
        }
        sides = sortedSides.toSet()
    }

    val isEquilateral: Boolean = sides.size == 1
    val isIsosceles: Boolean = sides.size <= 2
    val isScalene: Boolean = sides.size == 3
}
